javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+        }
+
+        @TearDown
+        public void tearDown() {
+            keysRandom = null;
+            keysRandom = null;
+            entriesRandom = null;
+            slangTreeMap = null;
+            javaTreeMap = null;
+            scalaTreeMap = null;
+        }
+
+        // Valid keys are only even numbers
+        int nextValidRandomKey() {
+            return keysRandom.get(random.nextInt(numItems));
+        }
+
+        int nextRandomKey() {
+            return random.nextInt(maxKey);
+        }
+
+        Array<Integer> randomize(Array<Integer> list) {
+            java.util.List<Integer> integers = list.toJavaList();
+            Collections.shuffle(integers);
+            return Array.ofAll(integers);
+        }
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Slang(BenchmarkState state) {
+        TreeMap<Integer, String> treeMap = TreeMap.empty();
+        for (Integer key : state.keysRandom) {
+            treeMap = treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Java(BenchmarkState state) {
+        java.util.TreeMap<Integer, String> treeMap = new java.util.TreeMap<>();
+        for (Integer key : state.keysRandom) {
+            treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Scala(BenchmarkState state) {
+        scala.collection.immutable.TreeMap<Integer, String> treeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+        for (Integer key : state.keysRandom) {
+            treeMap = treeMap.$plus(new scala.Tuple2<>(key, Integer.toString(key)));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Slang(BenchmarkState state) {
+        TreeMap<Integer, String> treeMap = TreeMap.empty();
+        for (Integer key : state.keysSorted) {
+            treeMap = treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Java(BenchmarkState state) {
+        java.util.TreeMap<Integer, String> treeMap = new java.util.TreeMap<>();
+        for (Integer key : state.keysSorted) {
+            treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Scala(BenchmarkState state) {
+        scala.collection.immutable.TreeMap<Integer, String> treeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+        for (Integer key : state.keysSorted) {
+            treeMap = treeMap.$plus(new scala.Tuple2<>(key, Integer.toString(key)));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object getValue_Slang(BenchmarkState state) {
```

Comment:
I found that by reducing the number of groups, the benchmark runs considerably faster (i.e. less warm-ups, forks and state creation), hence I decided to group related operations together and reduce number of groups. Get and Put are related in that they are complementary operations.

Please note that the `BenchmarkPerformanceReporter` is "intelligent" enough to recognise different implementations of the same operation. So you can group related operations 
into the same group, and they will not get mixed up in the performance report.

The `BenchmarkPerformanceReporter` expects the following method name convention:
```
void testoperation_implementation()
```

So if you have the following methods in the same group: `get_Java(), get_Scala(), get_Slang(), put_Java() and put_Slang()`, you will get a report like this one:

```
                                     10       1000     100000                                    
GetAndPut  get    Java/Slang      1.33x      1.22x      1.24x 
GetAndPut  get   Scala/Slang      1.21x      1.21x      1.58x 
GetAndPut  put    Java/Slang      1.21x      1.21x      1.58x 

```